secure/storage/src/vault.rs

```diff
@@ -44,14 +44,34 @@ impl VaultStorage {
         self.client.write_secret(key, key, &value.to_base64()?)?;
         Ok(())
     }
+
+    fn update_policy(
+        &self,
+        policy_name: &str,
+        key: &str,
+        capabilities: &[Capability],
+    ) -> Result<(), Error> {
+        let path = format!("secret/data/{}", key);
+        let mut vault_policy = self.client.read_policy(&policy_name).unwrap_or_default();
+        let vault_capabilities: Vec<vault::Capability> = capabilities
+            .iter()
+            .map(|c| match c {
+                Capability::Write => vault::Capability::Update,
+                Capability::Read => vault::Capability::Read,
+            })
+            .collect();
+        vault_policy.add_policy(&path, vault_capabilities);
+        self.client.create_policy(policy_name, &vault_policy)?;
```

Comment:
this feels like `update_policy` instead of `create_policy`?

Reply:
Maybe set policy... The documentation here is pretty poor as they call it create / update :/